src/app/components/navigation/home.rs

```diff
@@ -117,22 +147,51 @@ impl HomePane {
                 Box::new(now_playing),
             ],
             saved_playlists_model,
+            user_id,
+            listbox_fun,
+            row_activated_handler,
+            row_selected_handler,
         }
     }
 
-    pub fn connect_navigated<F: Fn() + 'static>(&self, f: F) {
-        let model = self.saved_playlists_model.clone();
-        self.listbox
-            .connect_row_activated(clone!(@weak self.stack as stack => move |_, row| {
+    pub fn connect_navigated(&mut self) {
+        let playlist_model = self.saved_playlists_model.clone();
+        let f = self.listbox_fun.clone();
```

Comment:
is it not possible to keep the old way, am I missing something? (not storing the fun, just passing it as an arg to connect_navigated)